TestHelper.psm1

```diff
@@ -201,3 +204,209 @@ function Install-ResourceDesigner {
+
+    .EXAMPLE
+        $TestEnvironment = Inialize-TestEnvironment `
+            -DSCModuleName 'xNetworking' `
+            -DSCResourceName 'MSFT_xFirewall'
+            -TestType Integration
+            
+        This command will initialize the test enviroment for Integration testing
+        the MSFT_xFirewall DSC resource in the xNetworking DSC module.    
+#>
+function Initialize-TestEnvironment
+{
+    [OutputType([PSObject])]
+    [CmdletBinding()]
+    Param
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String] $DSCModuleName,
+        
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String] $DSCResourceName,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateSet('Unit','Integration')]
+        [String] $TestType
+    )
+    
+    Write-Verbose -Verbose (`
+        'Initializing Test Environment for {0} testing of {1} in module {2}.' `
+            -f $TestType,$DSCResourceName,$DSCModuleName)   
+    if ($TestType -eq 'Unit')
+    {
+        [String] $RelativeModulePath = "DSCResources\$DSCResourceName\$DSCResourceName.psm1"
+    }
+    else
+    {
+        [String] $RelativeModulePath = "$DSCModuleName.psd1"
+    }
+
+    # Unique Temp Working Folder - always gets removed on completion
+    # The tests can put anything in here and it will get cleaned up.
+    [String] $RandomFileName = [System.IO.Path]::GetRandomFileName()
+    [String] $WorkingFolder = Join-Path -Path $env:Temp -ChildPath "$DSCResourceName_$RandomFileName" 
+    # Create the working folder if it doesn't exist (it really shouldn't anyway)
+    if (-not (Test-Path -Path $WorkingFolder))
+    {
+        New-Item -Path $WorkingFolder -ItemType Directory
+    } 
+    
+    # The folder where this module is found
+    [String] $moduleRoot = $pwd
+
+    # The folder that all tests will find this module in
+    [string] $modulesFolder = Resolve-Path -Path (Join-Path -Path $pwd -Child '..')   
+        
+    # Import the Module
+    $Splat = @{
+        Path = $moduleRoot
+        ChildPath = $RelativeModulePath
+        Resolve = $true
+        ErrorAction = 'Stop'
+    }
+    $DSCModuleFile = Get-Item -Path (Join-Path @Splat)
+    
+    # Remove all copies of the module from memory so an old one is not used.
+    if (Get-Module -Name $DSCModuleFile.BaseName -All)
+    {
+        Get-Module -Name $DSCModuleFile.BaseName -All | Remove-Module
+    }
+    
+    # Import the Module to test.
+    Import-Module -Name $DSCModuleFile.FullName -Force
+    
+    # Set the PSModulePath environment variable so that the module path this module is in
+    # appears first because the LCM will use this path to try and locate modules when integration
+    # tests are called. This is to ensure the correct module is tested.
+    [String] $OldModulePath = $env:PSModulePath
+    [String] $NewModulePath = $OldModulePath
+    if (($NewModulePath).Split(';') -ccontains $modulesFolder)
```

Comment:
I figured it out - the problem is actually caused by the Issue #12. This is the resulting 
```powershell
$ENV:PSModulePath after running the code:
    [String] $OldModulePath = $env:PSModulePath
    [String] $NewModulePath = $OldModulePath
    if (($NewModulePath).Split(';') -ccontains $modulesFolder)
    {
        # Remove the existing module from the module path if it exists
        $NewModulePath = ($NewModulePath -split ';' | Where-Object {$_ -ne $modulesFolder}) -join ';'
    }
    $NewModulePath = "$modulesFolder;$NewModulePath"

    Write-Verbose -Verbose (`
        'Updating $ENV:PSModulesFolder to "{0}".' `
            -f $NewModulePath)   

    $env:PSModulePath = $NewModulePath
    [System.Environment]::SetEnvironmentVariable('PSModulePath',$NewModulePath,[System.EnvironmentVariableTarget]::Machine)
```

![2015-12-16_11-31-06](https://cloud.githubusercontent.com/assets/7589164/11826240/010d5c08-a3e9-11e5-895d-53eb9d932fe3.png)

I completely see what @tysonjhayes is saying now and understand why the original code you wrote moved the module to the Program Files folder - as it does eliminate the problem caused by Issue #12.

I could put a fix in to ensure the current path is also removed from the $ENV:PSModulesPath - but if Issue #12 is fixed then this isn't required.